core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -403,6 +405,16 @@ private synchronized void initServer() throws IOException {
         .setProperties(properties)
         .setParameters(parameters)
         .build();
+    super.registerMetrics();
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricKey.CLUSTER_LEADER_INDEX.getName(),
+        () -> getLeaderIndex());
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricKey.MASTER_ROLE_ID.getName(),
+        () -> getRoleId());
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricKey.CLUSTER_LEADER_ID.getName(),
+        () -> getLeaderId());
```

Comment:
Will metrics added on the `RaftJournalSystem` add performance impact?

Reply:
`super.registerMetrics();` has a gauge on journal sequence, and i guess that's fine on performance. But if you can accept a lag, use `MetricsSystem.registerCachedGaugeIfAbsent()`.

Reply:
`registerCachedGaugeIfAbsent ` looks good, but 10 miniutes lag is so long.